save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt

```diff
@@ -201,7 +248,7 @@ class DockerService(private val configProperties: ConfigProperties) {
         )
         // todo: un-hardcode script
         if (File(resourcesPath, "examples/kotlin-diktat/run.sh").exists()) {
-            val cliCommand = "bash ./examples/kotlin-diktat/run.sh || ./$SAVE_CLI_EXECUTABLE_NAME"
+            val cliCommand = "bash ./examples/kotlin-diktat/run.sh || ./$SAVE_CLI_EXECUTABLE_NAME$saveCliExecFlags"
```

Comment:
Basically no, since we operate with different flags, and them should just be merged.

However the one thing can ruin logic: standard test suites are located in separate directory - `standardTestSuiteDir`, so we need to pass it to the `cliArg` (previously as `testRootPath`, now as `testFile.first()`). 

But in `TestService` we already pass another dir as `testFiles.first()`. In this case the only thing should be matter: `agentPropertiesFile` i.e. `standardTestSuiteDir` should be read before `${File(it.propertiesRelativePath).parent}` from `TestService`, then `{File(it.propertiesRelativePath).parent}` will be ignored by SAVE

Reply:
So we will basically have cmd like `./save <standardSuiteDir> <some dir filled by TestService> <list of test names>` and SAVE will treat `<some dir filled by TestService>` as test name and skip it as it's not existent? Seems not very good to rely on this behavior, we might make it stricter in the future and SAVE will start to fail in such cases

Reply:
I looked at the code more detally:

From `TestService.kt` the value of `testFile.first() == ${File(it.propertiesRelativePath).parent}`, hands in the `getNewTestsIds()` as `batch.suitesToArgs.values.first()`. 

In `getNewTestsIds()`  we also append some another flags, like `--report-type` etc.

The `getNewTestsIds()` function is called in `startHeartbeats()` from `SaveAgent.kt` and we use all arguments here as `heartbeatResponse.cliArgs`.

After some chain of function calls, we come to the `runSave`

In `runSave` we execute SAVE as following:
`.exec(config.cliCommand.let { if (cliArgs.isNotEmpty()) "$it $cliArgs" else it }`

And `./$SAVE_CLI_EXECUTABLE_NAME$saveCliExecFlags` from `DockerService.kt` which I add in this PR will received in  `runSave` as `cliCommand`.

`cliCommand` is located before `cliAgrs` from TestService.kt, so all should be okay,
the result command should looked like

    ./save standardTestSuiteDir --include-suites [list of standard suites] ${File(it.propertiesRelativePath).parent} --report-type json --result-output file ...

SAVE will parse testFiles as [standardTestSuiteDir, ${File(it.propertiesRelativePath).parent}] but since for now we doesn't support in SAVE to execute multiple dirs at the same time, the `${File(it.propertiesRelativePath).parent}` will be skiped

Reply:
Agree, that's not really good. Also it's very easy to forget about this trick.

However, for now, I can't see some easy ways to catch this situation.

The logic of SAVE arguments quite scattered between `TestService` `AgentService` and `DockerService`, may be it will be good to do some refactoring later? And after what we probably can process this trick in more correct way

Reply:
The idea behind setting some flags per batch of tests inside `TestService` was that we may need different settings for different suites. Now, however, we face quite the opposite: we need some static flags per execution. We already have this comment about the former:
https://github.com/cqfn/save-cloud/blob/582e3da2ea0f2dcc465fda281dd59df6e37be736/save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/AgentService.kt#L52-L59
Also, report type and output type are also static flags and can be set from `agent.properties` file.
So what we can do in the future is to remove `suitesToArgs`, set all static flags in `DockerService` via agent's properties file and only use dynamic flags in `AgentService` when setting names of tests.

Reply:
#257